mitodl/odl-video-service
# Per-user moira list cache

#### What are the relevant tickets?
- Fixes #518 

#### What's this PR do?
- Implements a per-user cache of moira lists that a user does and does not have access to, in order to reduce the frequency of moira service calls.

#### How should this be manually tested?
- There should be no obvious change to the user in how permissions work.
- Create a collection with `odl-engineering` as the view or admin moira list.
- In django admin, create a staff (but non-superuser) account for someone on the `odl-engineering` moira list (most likely your own MIT email)- username and email should both be `<name>@mit.edu`
- In an incognito browser window, log in as that non-superuser and go to a video detail page from that collection above.  You should be able to view it.  If `odl-engineering` is the admin moira list, you should be able to go to the collection detail page.  
- Add the following to your `hosts` file:
  ```
  127.0.0.1       moiraws.mit.edu
  ```
- Now go to the other video detail pages in the collection as the non-superuser.  The video detail page should load without any errors because the moira lists you have access to should already be cached.
- Check out the master branch from git and try the above step again.  Now you should get a `MoiraException`.
- Remove the change to your `hosts` file.
